cli/testdataseeder.php

```diff
@@ -115,15 +120,40 @@ function create_quiz_submitted($table) {
 }
 
 function create_forum_post($table) {
-    $str = "\\mod_forum\\event\\discussion_created','mod_forum','created','discussion','forum_discussions',1,'c',2,4947,70,7,3,7,NULL,0,'a:1:{s:7:\"forumid\";i:2;}',".time().",'web','172.19.0.1',NULL,'0','0'";
+    $str = "'\\mod_forum\\event\\discussion_created','mod_forum','created','discussion','forum_discussions',1,'c',2,4947,70,7,3,7,NULL,0,'a:1:{s:7:\"forumid\";i:2;}',".time().",'web','172.19.0.1',NULL,'0','0'";
     insert_row($table, $str);
 }
 
-function create_test_data($table) {
+function create_assignment_submitted($table) {
+    $str = "'\\mod_assign\\event\\assessable_submitted','mod_assign','submitted','assessable','assign_submission',2,'u',2,4948,70,8,3,7,NULL,0,'a:1:{s:19:\"submission_editable\";b:1;}',".time().",'web','172.19.0.1',NULL,'0','0'";
+    insert_row($table, $str);
+}
+
+function create_assignment_graded($table) {
+    $str = "'\\mod_assign\\event\\submission_graded','mod_assign','graded','submission','assign_grades',1,'u',1,4948,70,8,2,7,3,0,'N;',".time().",'web','172.19.0.1',NULL,'0','0'";
+    insert_row($table, $str);
+}
+
+/**
+ * Create a number of rows in the table.
+ * Add one row for each type, at least 10, pad the rest out with rows
+ * minus the number of message types.
+ *
+ * @param string $table tablename
+ * @param int $rows number of rows
+ */
+function create_test_data($table, $rows) {
+    if ($rows < 10) {
+        $rows = 10;
+    }
+    $course_viewed = $rows - 8;
```

Comment:
What is the purpose of the $row's variable? Also if rows is < 10 then $course_viewed always equals 2? this is a bit confusing with what's in the doc block as well

Reply:
If the number of rows is less than 10 it is so that we get at least one message of each type and 2 course_viewed rows. It's just so the user doesn't specify 0 rows for example.